packages/tsx-editor/src/transpiler/transpile.tsx

```diff
@@ -0,0 +1,37 @@
+import * as monaco from 'monaco-editor';
+import { ITranspiledOutput, IEvalCode } from './transpile.types';
+import { TypeScriptWorker, EmitOutput } from './monacoTypescriptWorker';
+import { ITextModel } from '../components/Editor.types';
+
+export function transpile(model: ITextModel): Promise<ITranspiledOutput> {
+  const ret = new Promise<ITranspiledOutput>(resolve => {
+    monaco.languages.typescript.getTypeScriptWorker().then(_worker => {
+      _worker(model.uri).then((worker: TypeScriptWorker) => {
+        worker.getEmitOutput(model.uri.toString()).then((output: EmitOutput) => {
+          const transpiledOutput: ITranspiledOutput = { error: undefined, outputString: undefined };
+          if (output.outputFiles[0]) {
+            transpiledOutput.outputString = output.outputFiles[0].text;
+          } else {
+            transpiledOutput.error = 'Could not transpile code';
+          }
+          resolve(transpiledOutput);
+        });
+      });
+    });
+  });
+  return ret;
+}
+
+export function _evalCode(code: string): IEvalCode {
+  const output: IEvalCode = {
+    eval: undefined,
+    error: undefined
+  };
+  try {
```

Comment:
I don't think you need to return the value from the eval() call--usually what you're interested in is the side effects of running the code, not whatever the code returns (if anything). I'd recommend updating the function to return undefined (implicitly) in the success case, and an error string in the error case.

```ts
export function evalCode(): string | undefined
  try {
    // tslint:disable:no-eval
    eval(code);
  } catch (ex) {
    return ex.message;
  }
}
```